File: src/main/java/axirassa/ioc/IocIntegrationTestRunner.java

package axirassa.ioc;

import org.apache.tapestry5.ioc.Registry;
import org.apache.tapestry5.ioc.RegistryBuilder;
import org.junit.runners.BlockJUnit4ClassRunner;
import org.junit.runners.model.InitializationError;

public class IocIntegrationTestRunner extends BlockJUnit4ClassRunner {

	private final Registry registry;


	public IocIntegrationTestRunner(Class<?> classObject) throws InitializationError {
		super(classObject);

		RegistryBuilder builder = IocStandalone.initRegistryBuilder();
		builder.add(HibernateTestingModule.class);

		registry = builder.build();
		registry.performRegistryStartup();
	}


	@Override
	public Object createTest() {
		Object test = registry.autobuild(getTestClass().getJavaClass());
		return test;
	}

}

File: src/main/java/axirassa/ioc/IocIntegrationTestRunner.scala

package axirassa.ioc

import org.apache.tapestry5.ioc.Registry
import org.junit.runners.BlockJUnit4ClassRunner
class IocIntegrationTestRunner(classObject : Class[_]) extends BlockJUnit4ClassRunner(classObject) {

    var registry : Registry = _
    val builder = IocStandalone.initRegistryBuilder()
    
    builder.add(classOf[HibernateTestingModule])

    registry = builder.build()
    registry.performRegistryStartup()

    override def createTest() = {
        registry.autobuild(getTestClass().getJavaClass()).asInstanceOf[Object]
    }
}
